libs/agno/agno/tools/apify.py

```
@@ -1,121 +1,112 @@
-            self.register(self.web_scrapper)
-
-    def website_content_crawler(self, urls: List[str], timeout: Optional[int] = 60) -> str:
         """
-        Crawls a website using Apify's website-content-crawler actor.
+        Initialize ApifyTools with specific actors.
 
-        :param urls: The URLs to crawl.
-        :param timeout: The timeout for the crawling.
-
-        :return: The results of the crawling.
+        Args:
+            actors: Single actor ID as string or list of actor IDs to register as individual tools
+            apify_api_token: Apify API token (defaults to APIFY_API_TOKEN env variable)
+            register_generic: Whether to register the generic run_actor method
         """
-        if self.api_key is None:
-            return "No API key provided"
-
-        if urls is None:
-            return "No URLs provided"
-
-        client = ApifyClient(self.api_key)
-
-        log_debug(f"Crawling URLs: {urls}")
-
-        formatted_urls = [{"url": url} for url in urls]
-
-        run_input = {"startUrls": formatted_urls}
-
-        run = client.actor("apify/website-content-crawler").call(run_input=run_input, timeout_secs=timeout)
-
-        results: str = ""
-
-        for item in client.dataset(run["defaultDatasetId"]).iterate_items():
-            results += "Results for URL: " + item.get("url") + "\n"
-            results += item.get("text") + "\n"
-
-        return results
-
-    def web_scrapper(self, urls: List[str], timeout: Optional[int] = 60) -> str:
+        super().__init__(name="ApifyTools")
+        
+        # Get API token from args or environment
+        self.apify_api_token = apify_api_token or os.getenv('APIFY_API_TOKEN')
+        if not self.apify_api_token:
+            raise ValueError("APIFY_API_TOKEN environment variable or apify_api_token parameter must be set")
+        
+        self.client = ApifyClient(self.apify_api_token)
+        
+        # Register specific actors if provided
+        if actors:
```

What happens if `actors` is None?
Does it make sense to do nothing? Should `actors` be required args or raise an error if not provided?

@MQ37 what do you think?

---

For me - even though it doesn't make much sense from a usability perspective it could still make sense to initialize this class.
It could theoretically allow for actor registration after initialization - lazy adding of Actors with register_actor method.
It also checks the Api key beforehand which I believe follows common Python pattern of optional parameters.

---

I think the actors should be required